src/node/api_clients/rest.py

```diff
@@ -52,6 +52,10 @@ def ping_filter_subscriptions(self, request_id):
         ping_subscriptions_response = self.rest_call("get", f"filter/v2/subscriptions/{quote(request_id, safe='')}")
         return ping_subscriptions_response.json()
 
-    def get_filter_messages(self, content_topic):
-        get_messages_response = self.rest_call("get", f"filter/v2/messages/{quote(content_topic, safe='')}")
+    def get_filter_messages(self, content_topic, pubsub_topic=None):
+        if pubsub_topic is not None:
+            endpoint = f"filter/v2/messages/{quote(pubsub_topic, safe='')}/{quote(content_topic, safe='')}"
+        else:
+            endpoint = f"filter/v2/messages/{quote(content_topic, safe='')}"
```

Comment:
An alternate approach I realised, both are fine by me :)
```suggestion
        endpoint = f"filter/v2/messages/{quote(content_topic, safe='')}"
        if pubsub_topic is not None:
            endpoint = f"{endpoint}/{quote(content_topic, safe='')}"  # or endpoint += f"/{quote(content_topic, safe='')}"
```

Reply:
Will not work this way because if pubsub_topic is not None we have `{quote(pubsub_topic, safe='')}` before `{quote(content_topic, safe='')}`
The way you suggested would work if **pubsub_topic** would come after **content_topic** param

Reply:
Oh, I misread that piece of code, got it